source/parsing/Preprocessor.cpp

```diff
@@ -98,6 +99,26 @@ void Preprocessor::pushSource(std::string_view source, std::string_view name) {
 void Preprocessor::pushSource(SourceBuffer buffer) {
     SLANG_ASSERT(buffer.id);
 
+    if (!options.keywordMapping.empty()) {
```

Comment:
This is very confusing; why do we sometimes use the keywordMapping and sometimes instead rely on the buffer's forcedKeywordVersion? It seems like you can remove all of the SourceLoader and forcedKeywordVersion stuff and just rely on the PreprocessorOptions to map buffers.